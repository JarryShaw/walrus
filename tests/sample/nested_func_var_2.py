def foo():
    x = 1
    def bar():
        x = 2
        def baz():
            nonlocal x
            print(x := 777)
        baz()
        print(x)
    bar()
    print(x)
foo()


def foo():
    (x := 1)
    def bar():
        nonlocal x
        (x := 2)
        def baz():
            nonlocal x
            (x := 3)
        baz()
        print(x)
    bar()
    print(x)
foo()
