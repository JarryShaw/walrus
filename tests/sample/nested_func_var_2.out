777
777
1
3
3
